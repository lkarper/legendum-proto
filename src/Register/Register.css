.Register__section {
    display: flex;
    flex-direction: column;
    margin: 0;
    padding: 10px;
    align-items: center;
    font-family: 'EB Garamond', serif;
}

.Register__section h2 {
    border-bottom: 1px solid black;
}

@media screen and (min-width: 1000px) {
    .Register__section h2 {
        font-size: 36px;
    }
}
