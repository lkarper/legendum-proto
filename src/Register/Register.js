import React, { useState } from 'react';
import AuthApiService from '../services/auth-api-service';
import RegistrationForm from '../RegistrationForm/RegistrationForm';

const Register = (props) => {
    const [userName, setUserName] = useState('');
    const [password, setPassword] = useState('');
    const [reenterPassword, setReenterPassword] = useState('');
    const [displayName, setDisplayName] = useState('');
    const [userNameError, setUserNameError] = useState(null);
    const [passwordErrorMessage, setPasswordErrorMessage] = useState([]);
    const [reenterPasswordError, setReenterPasswordError] = useState(null);
    const [displayNameError, setDisplayNameError] = useState(null);
    const [passwordError, setPasswordError] = useState({
        tooShort: true,
        tooLong: false,
        endSpaces: false,
        upperCase: true,
        lowerCase: true,
        number: true,
        specialChar: true,
    });
    const [apiError, setApiError] = useState(null);

    const handleSubmit = (event) => {
        event.preventDefault();
        AuthApiService.postUser({
            user_name: userName,
            password,
            display_name: displayName
        })
            .then(user => {
                setUserName('');
                setPassword('');
                setDisplayName('');
                setApiError(null);
                props.history.push('/login');
            })
            .catch(res => {
                setApiError(res.error);
            });
    }

    return (
        <>
            <RegistrationForm 
                propsObject={{
                    userName,
                    password,
                    reenterPassword,
                    displayName,
                    userNameError,
                    passwordErrorMessage,
                    reenterPasswordError,
                    displayNameError,
                    passwordError,
                    setUserName,
                    setPassword,
                    setReenterPassword,
                    setDisplayName,
                    setUserNameError,
                    setPasswordErrorMessage,
                    setReenterPasswordError,
                    setDisplayNameError,
                    setPasswordError,
                    handleSubmit,
                }}
            />
            {apiError ? 
                <div role="alert"> 
                    <h2>Error</h2>
                    <p>New user could not be created: {apiError}</p>
                </div>
                : ''
            }
        </>
    );
}

export default Register;