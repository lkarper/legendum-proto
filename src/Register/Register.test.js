import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Register from './Register';

describe('Register component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Register />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
