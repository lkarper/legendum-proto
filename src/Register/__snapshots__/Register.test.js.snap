// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Register
      history={
        Object {
          "push": [Function],
        }
      }
    >
      <section
        className="Register__section"
      >
        <h2>
          Register
        </h2>
        <RegistrationForm
          displayName=""
          displayNameError="Displayname required."
          handleSubmit={[Function]}
          password=""
          passwordError={
            Object {
              "endSpaces": false,
              "lowerCase": true,
              "number": true,
              "specialChar": true,
              "tooLong": false,
              "tooShort": true,
              "upperCase": true,
            }
          }
          passwordErrorMessage={
            Array [
              "Password must be at least 8 characters in length.",
              "Password must contain at least one lowercase character.",
              "Password must contain at least one uppercase character.",
              "Password must contain at least one number.",
              "Password must contain at least one special character.",
            ]
          }
          reenterPassword=""
          reenterPasswordError=""
          setDisplayName={[Function]}
          setDisplayNameError={[Function]}
          setPassword={[Function]}
          setPasswordError={[Function]}
          setPasswordErrorMessage={[Function]}
          setReenterPassword={[Function]}
          setReenterPasswordError={[Function]}
          setUserName={[Function]}
          setUserNameError={[Function]}
          userName=""
          userNameError="Username required."
        >
          <form
            className="RegistrationForm__form"
            onSubmit={[Function]}
          >
            <div>
              <label
                htmlFor="user-name"
              >
                Username:
              </label>
              <input
                aria-describedby="username-required"
                autoComplete="username"
                id="user-name"
                name="user-name"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <ValidateUserName
              setUserNameError={[Function]}
              userName=""
              userNameError="Username required."
            >
              <div
                role="alert"
              >
                <p
                  id="username-required"
                  style={
                    Object {
                      "color": "darkred",
                    }
                  }
                >
                  Username required.
                </p>
              </div>
            </ValidateUserName>
            <div>
              <label
                htmlFor="password"
              >
                Password:
              </label>
              <input
                aria-describedby="password-error-message"
                autoComplete="new-password"
                id="password"
                name="password"
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
            </div>
            <ValidatePassword
              password=""
              passwordError={
                Object {
                  "endSpaces": false,
                  "lowerCase": true,
                  "number": true,
                  "specialChar": true,
                  "tooLong": false,
                  "tooShort": true,
                  "upperCase": true,
                }
              }
              passwordErrorMessage={
                Array [
                  "Password must be at least 8 characters in length.",
                  "Password must contain at least one lowercase character.",
                  "Password must contain at least one uppercase character.",
                  "Password must contain at least one number.",
                  "Password must contain at least one special character.",
                ]
              }
              setPasswordError={[Function]}
              setPasswordErrorMessage={[Function]}
            >
              <PasswordChecklist
                passwordError={
                  Object {
                    "endSpaces": false,
                    "lowerCase": true,
                    "number": true,
                    "specialChar": true,
                    "tooLong": false,
                    "tooShort": true,
                    "upperCase": true,
                  }
                }
              >
                <div
                  className="PasswordChecklist__container"
                >
                  <p>
                    Password must:
                  </p>
                  <ul>
                    <li>
                      Be at least 8 characters in length 
                      <span
                        className="PasswordChecklist__x"
                      >
                        ✖
                      </span>
                    </li>
                    <li>
                      Be no more than 72 characters in length 
                      <span
                        className="PasswordChecklist__check"
                      >
                        ✔
                      </span>
                    </li>
                    <li>
                      Not begin or end with a space 
                      <span
                        className="PasswordChecklist__check"
                      >
                        ✔
                      </span>
                    </li>
                    <li>
                      Contain at least one lowercase letter 
                      <span
                        className="PasswordChecklist__x"
                      >
                        ✖
                      </span>
                    </li>
                    <li>
                      Contain at least one uppercase letter 
                      <span
                        className="PasswordChecklist__x"
                      >
                        ✖
                      </span>
                    </li>
                    <li>
                      Contain at least one number 
                      <span
                        className="PasswordChecklist__x"
                      >
                        ✖
                      </span>
                    </li>
                    <li>
                      Contain at least one special character (e.g. !, @, #, $, etc.) 
                      <span
                        className="PasswordChecklist__x"
                      >
                        ✖
                      </span>
                    </li>
                  </ul>
                </div>
              </PasswordChecklist>
              <div
                className="ValidatePassword__error-container"
                role="alert"
              >
                <p
                  id="password-error-message"
                >
                  Password must be at least 8 characters in length. Password must contain at least one lowercase character. Password must contain at least one uppercase character. Password must contain at least one number. Password must contain at least one special character.
                </p>
              </div>
            </ValidatePassword>
            <div>
              <label
                htmlFor="reenter-password"
              >
                Password:
              </label>
              <input
                aria-describedby="reenter-password-error"
                autoComplete="new-password"
                id="reenter-password"
                name="reenter-password"
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
            </div>
            <ValidateReenterPassword
              password=""
              reenterPassword=""
              reenterPasswordError=""
              setReenterPasswordError={[Function]}
            >
              <div
                role="alert"
              >
                <p
                  id="reenter-password-error"
                >
                  Passwords match.
                </p>
              </div>
            </ValidateReenterPassword>
            <div>
              <label
                htmlFor="display-name"
              >
                Display name:
              </label>
              <input
                aria-describedby="display-name-required"
                id="display-name"
                name="display-name"
                onChange={[Function]}
                required={true}
                type="input"
                value=""
              />
            </div>
            <ValidateDisplayName
              displayName=""
              displayNameError="Displayname required."
              setDisplayNameError={[Function]}
            >
              <div
                role="alert"
              >
                <p
                  id="display-name-required"
                  style={
                    Object {
                      "color": "darkred",
                    }
                  }
                >
                  Displayname required.
                </p>
              </div>
            </ValidateDisplayName>
            <button
              className="RegistrationForm__submit button"
              disabled={true}
              type="submit"
            >
              Register
            </button>
          </form>
        </RegistrationForm>
        <div
          className="Register__alert-div"
          role="alert"
        />
      </section>
    </Register>
  </Router>
</BrowserRouter>
`;
